bin/admin/check_uid_gid_collisions.pl

```diff
@@ -0,0 +1,328 @@
+#! /usr/bin/env perl
+# vim: set filetype=perl ts=4 sw=4 sts=4 et:
+use strict;
+use warnings;
+use 5.010;
+
+use Getopt::Long;
+
+use File::Basename;
+use lib dirname(__FILE__) . '/../../lib/perl';
+
+my $getOptionsOk = GetOptions(
+    "master-passwd=s" => \my $masterPasswdFile,
+    "local-passwd=s"  => \my $localPasswdFile,
+    "master-group=s"  => \my $masterGroupFile,
+    "local-group=s"   => \my $localGroupFile,
+    "output=s"        => \my $outputFile,
+    "offset=i"        => \my $offset,             # undocumented on purpose
+);
+
+if (!$getOptionsOk) {
+    print "Failed to parse options (see above), aborting.\n";
+    exit(1);
+}
+
+$localPasswdFile //= '/etc/passwd';
+$localGroupFile  //= '/etc/group';
+$offset          //= 50_000_000;
+
+if ($offset < 10_000) {
+    die "Offset is too low ($offset)";
+}
+
+if (!$masterGroupFile || !$masterPasswdFile || !$outputFile) {
+    die "Usage: $0 --master-passwd PATH --master-group PATH --output FILE [--local-passwd PATH --local-group PATH]";
+}
+
+if (-e $outputFile) {
+    die "Output file '$outputFile' already exists!";
+}
+
+my (%local, %master);
+read_pwgr_file($masterPasswdFile, \%master, 'pw');
+read_pwgr_file($localPasswdFile,  \%local,  'pw');
+read_pwgr_file($masterGroupFile,  \%master, 'gr');
+read_pwgr_file($localGroupFile,   \%local,  'gr');
+
+sub read_pwgr_file {
+    my ($file, $hashref, $type) = @_;
+    my $fh;
+    if (!open($fh, '<', $file)) {
+        die("Couldn't open file '$file': $!");
+    }
+    $hashref->{$type} = {};
+  LINE: while (<$fh>) {
+        chomp;
+        my @fields = split /:/;
+
+        my $name = $fields[0];
+        my $id   = $fields[2];
+
+        next if $id == 0;
+
+        $hashref->{$type}{name_by_id}{$id}   = $name;
+        $hashref->{$type}{id_by_name}{$name} = $id;
+    }
+    close($fh);
+    return;
+}
+
+my $orphans;
+# first, report orphans
+handle_list($local{'pw'}, $master{'pw'}, 'pw', 0);
+handle_list($local{'gr'}, $master{'gr'}, 'gr', 0);
+if ($orphans) {
+    print "\nThere is at least one warning, see above.\n";
+    print "If you want to handle them, you may still abort now.\n";
+    print "Type 'YES' to proceed regardless.\n";
+    my $ans = <>;
+    chomp $ans;
+    if ($ans ne 'YES') {
```

Comment:
Suggestion: `uc($ans) eq 'YES'`

Reply:
Actually, I want it uppercase, to ensure the user acknowledges that they really know what they're doing (I stole the idea from `cryptsetup`)